### PR TITLE
STAR-57 - list for front

### PR DESCRIPTION
**Pull request by KrzysztofDziedziechEscolasoft**

### CODE REVIEW

File: src/Http/Requests/QuestionnaireFrontAnswerRequest.php [hunk 1]
```diff
@@ -9,6 +9,26 @@
...
+ *         type="integer",
+ *         description="question identified by id"
+ *     ),
+ *     @OA\Property(
+ *          property="answers.*.rate",
+ *          type="integer",
+ *          description="rate from 1 to 5"
+ *     )
+ * )
+ */
```
Comment (pa-cholek) [inline src/Http/Requests/QuestionnaireFrontAnswerRequest.php:31]:
```suggestion
/**
 * @OA\Schema(
 *     schema="QuestionnaireAnswerRequest",
 *     @OA\Property(
 *          property="answers",
 *          description="answers for questionnaire"
 *          @OA\Schema(
 *              type="array",
 *              @OA\Items(
 *                  type="object",
 *                  @OA\Property(
 *                      property="question_id",
 *                      type="integer",
 *                      description="question identified by id",
 *                  ),
 *                  @OA\Property(
 *                      property="rate",
 *                      type="integer",
 *                      description="rate from 1 to 5",
 *                  ),
 *              )
 *          )
 *     ),
 * )
 */
```

File: src/Repository/QuestionnaireRepository.php [hunk 2]
```diff
@@ -16,12 +16,24 @@ public function model(): string
...
+        ?int $perPage = null,
+        string $orderDirection = 'asc',
+        string $orderColumn = 'id'
+    ): LengthAwarePaginator {
+        $query = $this->allQuery($search);
+        if (isset($search['model_type_id'])) {
+            $query->select('questionnaires.*')
+                ->join('questionnaire_models', 'questionnaire_models.questionnaire_id', '=', 'questionnaires.id')
+                ->where('questionnaire_models.model_type_id', '=', $search['model_type_id'])
+                ->where('questionnaire_models.model_id', '=', $search['model_id']);
```
Comment (pa-cholek) [inline src/Repository/QuestionnaireRepository.php:33]:
```suggestion
            $query->whereHas('questionnaire_models', fn(Builder $query) => $query->where('model_type_id', $search['model_type_id'])->where('model_id', $search['model_id']);
```

File: src/Services/QuestionService.php [hunk 3]
```diff
@@ -30,8 +32,8 @@ public function deleteQuestion(Question $question): bool
             });
 
             return true;
-        } catch (\Exception $err) {
-            return false;
+        } catch (Exception $err) {
+            throw new UnprocessableEntityHttpException(__('Question deleted failed'));
```
Comment (pa-cholek) [inline src/Services/QuestionService.php:36]: Why catch here if all you are doing is throwing again? And throwing Http exception from Service and not Request/Controller is probably bad, what if this service was used in a Job/Queue/EventListener (something that happens outside of HTTP context)

**Review by pa-cholek — CHANGES_REQUESTED** [review]
Small changes requested

File: src/Services/QuestionService.php (hunk 3 above)
Comment (KrzysztofDziedziechEscolasoft) [reply src/Services/QuestionService.php]: done

**Review by qunabu — APPROVED** [review]

**Pull request merged**